src/shogun/lib/SGVector.cpp

```diff
@@ -114,40 +114,48 @@ SGVector<T>::~SGVector()
 
 template <class T>
 SGVector<T>::SGVector(EigenVectorXt& vec)
-: SGReferencedData(false), vector(vec.data()), vlen(vec.size())
+: SGReferencedData(false), vector(vec.data()), vlen(vec.size()), gpu_vector(NULL)
 {
 
 }
 
 template <class T>
 SGVector<T>::SGVector(EigenRowVectorXt& vec)
-: SGReferencedData(false), vector(vec.data()), vlen(vec.size())
+: SGReferencedData(false), vector(vec.data()), vlen(vec.size()), gpu_vector(NULL)
 {
 
 }
 
 template <class T>
 SGVector<T>::operator EigenVectorXtMap() const
 {
+	if (on_gpu())
+		SG_SERROR("EigenRowVectorXtMap method not available for vectors on GPU.\n")
 	return EigenVectorXtMap(vector, vlen);
 }
 
 template <class T>
 SGVector<T>::operator EigenRowVectorXtMap() const
 {
+	if (on_gpu())
+		SG_SERROR("EigenRowVectorXtMap method not available for vectors on GPU.\n")
 	return EigenRowVectorXtMap(vector, vlen);
 }
 
 template<class T>
 void SGVector<T>::zero()
 {
+	if (on_gpu())
+		SG_SERROR("Zero method not available for vectors on GPU.\n")
```

Comment:
actually why not? :)
i know why in a way but is it really something that wouldn't like to have? :P